flash/data/auto_dataset.py

```diff
@@ -0,0 +1,115 @@
+from copy import deepcopy
+from inspect import signature
+from typing import Any, Callable, Optional, TYPE_CHECKING
+
+import torch
+from pytorch_lightning.trainer.states import RunningStage
+from pytorch_lightning.utilities.warning_utils import rank_zero_warn
+
+from flash.data.process import Preprocess
+
+if TYPE_CHECKING:
+    from flash.data.data_pipeline import DataPipeline
+
+
+class AutoDataset(torch.utils.data.Dataset):
+
+    FITTING_STAGES = ("train", "val")
+    STAGES = ("train", "test", "eval", "val", "predict")
```

Comment:
@tchaton I believe the STAGES need to be updated. Also, `validate`*.

Reply:
Note that the Enum itself is passed, so we could rely on it instead of its associated str